plugins/linux-capture/pipewire.c

```diff
@@ -405,19 +405,18 @@ static inline struct spa_pod *build_format(struct spa_pod_builder *b,
 
 	/* modifier */
 	if (modifier_count > 0) {
+		struct spa_pod_frame modifier_frame;
+
 		/* build an enumeration of modifiers */
 		spa_pod_builder_prop(b, SPA_FORMAT_VIDEO_modifier,
 				     SPA_POD_PROP_FLAG_MANDATORY |
 					     SPA_POD_PROP_FLAG_DONT_FIXATE);
-		spa_pod_builder_push_choice(b, &f[1], SPA_CHOICE_Enum, 0);
+		spa_pod_builder_push_choice(b, &modifier_frame, SPA_CHOICE_Enum,
+					    0);
 		/* modifiers from  an array */
-		for (uint32_t i = 0; i < modifier_count; i++) {
-			uint64_t modifier = modifiers[i];
-			spa_pod_builder_long(b, modifier);
-			if (i == 0)
-				spa_pod_builder_long(b, modifier);
-		}
-		spa_pod_builder_pop(b, &f[1]);
+		for (uint32_t i = 0; i < modifier_count; i++)
+			spa_pod_builder_long(b, modifiers[i]);
+		spa_pod_builder_pop(b, &modifier_frame);
```

Comment:
PipeWire Choice PODs have the structure: Preference, Item1, Item2, .... , ItemN
See eg:
https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/examples/sdl.h#L176
https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/spa/plugins/bluez5/a2dp-codec-ldac.c#L188

Maybe we should add a comment?

```suggestion
		for (uint32_t i = 0; i < modifier_count; i++)
		        // Note: PipeWire Choice PODs start with a preference followed by the list
			if (i == 0)
			        spa_pod_builder_long(b, modifiers[i]);
			spa_pod_builder_long(b, modifiers[i]);
		spa_pod_builder_pop(b, &modifier_frame);
```

Reply:
Thanks, I didn't know that. I've updated the patch so that the first modifier is still added, but outside the loop.